Editors/ViewModels/SchematicsViewModel.cs

```diff
@@ -147,24 +148,29 @@ private void SimulationThread(object state)
         }
 
         public bool IsSimulationRunning => simulation != null;
-        public IEnumerable<Entities.Entity> Entities
+        public List<Entities.Entity> Entities
         {
             get
             {
-                IEnumerable<Entities.Entity> allEntities = entities;
+                List<Entities.Entity> allEntities = entities.ToList();
                 if (IsSimulationRunning)
                 {
-                    return allEntities.Union(simulation.Entities);
+                    return (List<Entity>)allEntities.Union(simulation.Entities);
                 }
                 return allEntities;
             }
         }
+
+        public List<Entities.Entity> FinalEntities;
+        public Dictionary<Guid, int> entitiesIndexes = new Dictionary<Guid, int>();
```

Comment:
i would just make this a Dictionary<Guid, Entity> tho idk if the instanceguid is even unique for all entities, probably best to have it be some hash of the class and file guid. but yeah with this u dont need the list shit and hashset for entities, still need the referenceObjects one, tho idk for what thats used

Reply:
in deathstar_01, theres only about 5k unique instanceguids with a total of 181k entities.
how the old way did it was FirstOrDefault(), meaning it would just iterate through them, so the dictionary takes the first instance of that and sets it to the index and throws out any ones after that to match its old behavior.

Reply:
yeah i know how it was before, its just the way u did it seems to be over complicated. Also i think that its not correct as well since iirc 2 entities can have the same instance guid but are not the same, u should probably check for the classguid and fileguid to get a proper unique identifier

Reply:
if the old code didnt do that why should we now? theres got to have been a reason it just threw away sequential objects

Reply:
its been a while since i looked at that, but iirc it was a bug with entities loaded from different blueprints, which got fixed when using the file and class guid instead of the instance guid

Reply:
why wasnt this resolved already then? i saw in your build it had something similar to this